plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcIdentityCacheMapping.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+public class DefaultJdbcIdentityCacheMapping
```

Comment:
It looks way more complicated then apparently needed, while also being weaker than it could be.
e.g The fact that user sends in GCP's GS related extra credentials should not affect caching in a PostgreSQL connector.